src/components/Carousel.js

```diff
@@ -518,10 +521,15 @@ class Carousel extends Component {
             return null
         }
 
+
         return (
             <ul className="control-dots">
                 {Children.map(this.props.children, (item, index) => {
-                    return <li className={klass.DOT(index === this.state.selectedItem)} onClick={this.changeItem} value={index} key={index} />;
+                    const isSelected = index === this.state.selectedItem
+                    const onClick = this.changeItem
+                    return this.props.indicator ?
+                        this.props.indicator({onClick, value, isSelected}) :
```

Comment:
The `changeItem` method will look for `e.target.value` which forces the consumer of the library to implement their component in a very specific way. I think a better way to do this would be to pass `() => this.selectItem({ selectedItem: value })` as onClick for custom items instead. You can still pass the value to the indicator as it might be useful but it won't be required to make it work.